homeassistant/components/camera/webrtc.py

```diff
@@ -68,21 +77,83 @@ class RTCConfiguration(_RTCBaseModel):
 
 
 @dataclass(kw_only=True)
-class WebRTCClientConfiguration(_RTCBaseModel):
+class WebRTCClientConfiguration(DataClassDictMixin):
     """WebRTC configuration for the client.
 
     Not part of the spec, but required to configure client.
     """
 
+    class Config(BaseConfig):
+        """Mashumaro config."""
+
+        serialize_by_alias = True
+
     configuration: RTCConfiguration = field(default_factory=RTCConfiguration)
     data_channel: str | None = field(
         metadata=field_options(alias="dataChannel"), default=None
     )
+    get_candidates_upfront: bool = field(
+        metadata=field_options(alias="getCandidatesUpfront"), init=False, default=False
+    )
+
+
+@dataclass(frozen=True)
+class WebRTCSessionId(DataClassDictMixin):
+    """WebRTC session ID."""
+
+    session_id: str
+    type: str = field(default="session_id", init=False)
+
+
+@dataclass(frozen=True)
+class WebRTCAnswer(DataClassDictMixin):
+    """WebRTC answer."""
+
+    answer: str
+    type: str = field(default="answer", init=False)
+
+
+@dataclass(frozen=True)
+class WebRTCCandidate(DataClassDictMixin):
+    """WebRTC candidate."""
+
+    candidate: str
+    type: str = field(default="candidate", init=False)
+
+
+type WebRTCMessages = WebRTCAnswer | WebRTCCandidate | WebRTCSessionId
+type WebRTCSendMessage = Callable[[WebRTCMessages], None]
 
 
 class CameraWebRTCProvider(Protocol):
     """WebRTC provider."""
 
+    async def async_is_supported(self, stream_source: str) -> bool:
```

Comment:
Should this be async or can it be a callback ?

Reply:
I callback should be fine